league_api_integration.py

```diff
@@ -1,21 +1,52 @@
 """
-Integrates the bot with League Client Update (LCU) API.
-Note that the API is allowed to use, but not officially supported by Rito.
-The endpoints we use are stable-ish, and we do not expect them to change soonTM.
+Integrations with the Rito API to have the most reliable data where possible.
 """
 import time
 
 from loguru import logger
 from psutil import Process
 from psutil import process_iter
 import requests
+from requests import HTTPError
 
 # Potentially make this configurable in the future
 # to let the user select their preferred tft mode.
-from requests import HTTPError
-import urllib3
-
 TFT_NORMAL_GAME_QUEUE_ID = 1090
+ROOT_CERTIFICATE = """-----BEGIN CERTIFICATE-----
+MIIEIDCCAwgCCQDJC+QAdVx4UDANBgkqhkiG9w0BAQUFADCB0TELMAkGA1UEBhMC
+VVMxEzARBgNVBAgTCkNhbGlmb3JuaWExFTATBgNVBAcTDFNhbnRhIE1vbmljYTET
+MBEGA1UEChMKUmlvdCBHYW1lczEdMBsGA1UECxMUTG9MIEdhbWUgRW5naW5lZXJp
+bmcxMzAxBgNVBAMTKkxvTCBHYW1lIEVuZ2luZWVyaW5nIENlcnRpZmljYXRlIEF1
+dGhvcml0eTEtMCsGCSqGSIb3DQEJARYeZ2FtZXRlY2hub2xvZ2llc0ByaW90Z2Ft
+ZXMuY29tMB4XDTEzMTIwNDAwNDgzOVoXDTQzMTEyNzAwNDgzOVowgdExCzAJBgNV
+BAYTAlVTMRMwEQYDVQQIEwpDYWxpZm9ybmlhMRUwEwYDVQQHEwxTYW50YSBNb25p
+Y2ExEzARBgNVBAoTClJpb3QgR2FtZXMxHTAbBgNVBAsTFExvTCBHYW1lIEVuZ2lu
+ZWVyaW5nMTMwMQYDVQQDEypMb0wgR2FtZSBFbmdpbmVlcmluZyBDZXJ0aWZpY2F0
+ZSBBdXRob3JpdHkxLTArBgkqhkiG9w0BCQEWHmdhbWV0ZWNobm9sb2dpZXNAcmlv
+dGdhbWVzLmNvbTCCASIwDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBAKoJemF/
+6PNG3GRJGbjzImTdOo1OJRDI7noRwJgDqkaJFkwv0X8aPUGbZSUzUO23cQcCgpYj
+21ygzKu5dtCN2EcQVVpNtyPuM2V4eEGr1woodzALtufL3Nlyh6g5jKKuDIfeUBHv
+JNyQf2h3Uha16lnrXmz9o9wsX/jf+jUAljBJqsMeACOpXfuZy+YKUCxSPOZaYTLC
+y+0GQfiT431pJHBQlrXAUwzOmaJPQ7M6mLfsnpHibSkxUfMfHROaYCZ/sbWKl3lr
+ZA9DbwaKKfS1Iw0ucAeDudyuqb4JntGU/W0aboKA0c3YB02mxAM4oDnqseuKV/CX
+8SQAiaXnYotuNXMCAwEAATANBgkqhkiG9w0BAQUFAAOCAQEAf3KPmddqEqqC8iLs
+lcd0euC4F5+USp9YsrZ3WuOzHqVxTtX3hR1scdlDXNvrsebQZUqwGdZGMS16ln3k
+WObw7BbhU89tDNCN7Lt/IjT4MGRYRE+TmRc5EeIXxHkQ78bQqbmAI3GsW+7kJsoO
+q3DdeE+M+BUJrhWorsAQCgUyZO166SAtKXKLIcxa+ddC49NvMQPJyzm3V+2b1roP
+SvD2WV8gRYUnGmy/N0+u6ANq5EsbhZ548zZc+BI4upsWChTLyxt2RxR7+uGlS1+5
+EcGfKZ+g024k/J32XP4hdho7WYAS2xMiV83CfLR/MNi8oSMaVQTdKD8cpgiWJk3L
+XWehWA==
+-----END CERTIFICATE-----
+"""
+
+
+def write_root_certificate_to_file() -> None:
+    """
+    Writes the Riot root certificate to a file in the file system, because requests needs it to be a file.
+    We need the certificate for SSL connections to any of Riot's APIs.
+    """
+    with open(file="riotgames_root_certificate.pem", mode="w", encoding="UTF-8") as file:
```

Comment:
I understand the want to avoid appdata, however, for the users running the bot from the release executable, putting the cert in whatever directory this was run from is a bad practice and I'd prefer to avoid it.

I believe we also need to update the new logging system to use the appdata folder as well but that's out of the scope of this PR (though if it's a one-liner, that's fine IMO).